assessments/projects/opa/self-assessment.md

```diff
@@ -0,0 +1,563 @@
+
+### CII Best Practices
+
+Currently, OPA is at the passing criteria in the [Core Infrastructure Initiative (CII) best practices badging program](https://github.com/coreinfrastructure/best-practices-badge/blob/master/doc/criteria.md) ([security details here](https://bestpractices.coreinfrastructure.org/en/projects/1768)). OPA endeavours to reach the silver tier by following best
+practices and guidelines outlined by CII.
+
+### Case Studies
+
+OPA is a general-purpose policy engine that has multiple deployment and
+integration models and is used in production by more than 20 companies
+such as Netflix, Intuit, Capital One.
+
+#### REST API Authorization Example
+
+This example shows two simple rules that enforce an authorization policy on an
+API that serves salary data. In English, the policy says that
+*employees can see their own salary and the salary of any of their reports*.
+
+```ruby
+allow { 
+input.method = "GET" 
+input.path = ["salary", employee_id] 
+input.user = employee_id 
+}
+ 
+allow {
+input.method = "GET" 
+input.path = ["salary", employee_id] 
+input.user = data.management_chain[employee_id][_]
+}
+```
+
+The first rule allows employees to ***GET*** their own salary. The rule shows
+how you can use variables in rules. In that rule, ***employee_id*** is a
+variable that will be bound to the same value across the last two expressions.
+
+The second rule allows employees to ***GET*** the salary of their reports.
+The rule shows how you can access arbitrary context (e.g., JSON data) inside
+the policy. The data may be loaded into the policy engine (and cached) or it
+may be external and fetched dynamically.
+
+#### Kubernetes Admission Control Example
+
+This example shows how to enforce custom policies on Kubernetes objects
+using OPA. The below policy prevents Ingress objects in different namespaces
+from sharing the same hostname.
+
+```ruby
+deny[msg] {
+    input.request.kind.kind == "Ingress"
+    host := input.request.object.spec.rules[_].host
+    ingress := data.kubernetes.ingresses[other_ns][other_ingress]
+    other_ns != input.request.namespace
+    ingress.spec.rules[_].host == host
+    msg := sprintf("invalid ingress host %q (conflicts with %v/%v)", [host, other_ns, other_ingress])
+}
+```
+
+This example shows how OPA can be used to enforce admission control decisions in
+Kubernetes clusters on-the-fly without modifying or recompiling any Kubernetes
+components. Also the *deny* rule returns a set illustrating Rego’s ability to
+generate non-boolean policy decisions.
+
+More information on how OPA can be integrated with Kubernetes as an
+Admission Controller can be found [here](https://www.openpolicyagent.org/docs/latest/kubernetes-introduction/).
+
+### Related Projects / Vendors
+
+* OPA’s policy language REGO, allows policy decisions that are more than
```

Comment:
for consistent capitalisation should be "Rego".

Reply:
Done